src/language-js/comments/printer-methods.js

```diff
@@ -82,6 +82,14 @@ function willPrintOwnComments({ node, parent }) {
   );
 }
 
+function getGapRegex(enclosingNode, { parser }) {
+  if (parser === "flow" || parser === "babel-flow") {
+    // Example: (a /* b */ /* : c */)
+    //                gap ^^^^
+    return /^[\s(]*(\/\*(\s*::)?[\s(]*)?$/;
+  }
+}
```

Comment:
I remember there is one regexp for babel-ts? Where is it...

Reply:
https://github.com/prettier/prettier/commit/c1937cb1b8717d8fe2247de99e7031bd7fd7bab3#diff-112b7d78917580aa0cc6cc3ac0e641c6a9541c0ed4de574f05dba2ae51480ccaR914

Reply:
Ah, removed in #10457, fogot.

Reply:
Will `/^(\s|\(|\/\*\s*::)*$/` work? This is more readable to me.

Reply:
Can we use `isGap`? So we can use a function, the regexp is hard to read when it's long.

```js
function isGap(text) {
	text = text.replace(/[\s\(]/g, '')
	return !text || text === '/*::'
}
```